warden/x/intent/keeper/actions.go

```diff
@@ -143,3 +202,20 @@ func (k Keeper) AddAction(ctx context.Context, creator string, msg sdk.Msg, inte
 
 	return act, nil
 }
+
+// assert that the x/intent module account is the only signer of the message
+func (k Keeper) validateActionMsgSigners(msg sdk.Msg) error {
+	signers, _, err := k.cdc.GetMsgV1Signers(msg)
+	if err != nil {
+		return err
+	}
+	if len(signers) != 1 {
+		return types.ErrInvalidSigner
+	}
+
+	if sdk.AccAddress(signers[0]).String() != k.GetAuthority() {
```

Comment:
`GetAuthority` returns the module's authority, so do I understand correctly that only the module owner can add a new action because of this check? Otherwise, at what point is the user as signer replaced by the module account?

Reply:
nowhere, this is the signer address of the "wrapped" message and the clients will send transaction with the field already correctly set

A MsgNewAction will look like this:
```
auth_info:
  fee:
    amount: []
    gas_limit: "200000"
    granter: ""
    payer: ""
  signer_infos: []
  tip: null
body:
  extension_options: []
  memo: ""
  messages:
  - '@type': /warden.intent.MsgNewAction
    action_timeout_height: "0"
    creator: warden10kmgv5gzygnecf46x092ecfe5xcvvv9r870rq4
    message:
      '@type': /warden.warden.v1beta2.MsgAddSpaceOwner
      authority: warden19q4uhsls5d9g5jkx7qxzwm70vm8nw4arq2q3jh
      new_owner: ""
      space_id: "123"
  non_critical_extension_options: []
  timeout_height: "0"
signatures: []
```

in this example, `warden10kmgv5gzygnecf46x092ecfe5xcvvv9r870rq4` is my address and is the `creator` of the action

on the other hand, `warden19q4uhsls5d9g5jkx7qxzwm70vm8nw4arq2q3jh` is the address of the x/intent module, i.e. the "authority" that will execute the MsgAddSpaceOwner at a later point

does it make sense?

Reply:
Could there be a case where a different address can be used in the `authority` message field?

Reply:
If there is no such case, can we simplify the message by removing this field?

Reply:
yes, users can try to send a (for example) MsgAddSpaceOwner themselves, from their personal account, and we need to block this - MsgAddSpaceOwner can only be executed by the x/intent module account